sdk/mobile-center-push/src/main/java/com/microsoft/azure/mobile/push/Push.java

```diff
@@ -111,6 +117,29 @@ public static void setEnabled(boolean enabled) {
     }
 
     /**
+     * Enable firebase analytics collection.
+     *
+     * @param context the context to retrieve FirebaseAnalytics instance.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static void enableFirebaseAnalytics(@NonNull Context context) {
+        MobileCenterLog.debug(LOG_TAG, "Enabling firebase analytics collection.");
+        setFirebaseAnalyticsEnabled(context, true);
+    }
+
+    /**
+     * Enable or disable firebase analytics collection.
+     *
+     * @param context the context to retrieve FirebaseAnalytics instance.
+     * @param enabled <code>true</code> to enable, <code>false</code> to disable.
+     */
+    @SuppressWarnings("MissingPermission")
+    private static void setFirebaseAnalyticsEnabled(@NonNull Context context, boolean enabled) {
+        FirebaseAnalyticsUtils.setEnabled(context, enabled);
+        PreferencesStorage.putBoolean(PREFERENCE_KEY_ANALYTICS_ENABLED, enabled);
```

Comment:
To be clear:

* In every case call setAnalyticsCollectionEnabled(true) upon calling Push.enableFirebaseAnaltyics.
* If the method is called before start, store temp boolean to remember not to disable on start.